src/Utilities/TimerManager.cpp

```diff
@@ -74,6 +75,40 @@ TIMER* TimerManager<TIMER>::createTimer(const std::string& myname, timer_levels
   return t;
 }
 
+template<class TIMER>
+void TimerManager<TIMER>::push_timer(TIMER* t)
+{
+  // current_timer() can be nullptr when the stack was empty.
+  if (t == current_timer())
+  {
+    std::cerr << "Timer " << t->get_name()
+              << " instance is already at the top of the stack. "
+                 "start() is being called again. This often happens when stop() is not paired properly with start(). "
+              << "ScopedTimer uses RAII and managers timer start/stop much safer." << std::endl;
+    throw std::runtime_error("TimerManager push_timer error!");
+  }
+  else
+    CurrentTimerStack.push_back(t);
+}
+
+template<class TIMER>
+void TimerManager<TIMER>::pop_timer(TIMER* t)
+{
+  TIMER* stack_top = current_timer();
+  if (stack_top == nullptr)
+  {
+    std::cerr << "Timer stack pop failed on an empty stack! Requested \"" << t->get_name() << "\"." << std::endl;
+    throw std::runtime_error("TimerManager pop_timer error!");
+  }
+  else if (t != stack_top)
+  {
+    std::cerr << "Timer stack pop not matching push! "
+              << "Expect \"" << t->get_name() << "\" but \"" << stack_top->get_name() << "\" on the top." << std::endl;
```

Comment:
Possible change: "Timer stack pop does not match push"
            "Expecting ... but ... is on the top"